Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -772,7 +772,9 @@ extension TaskHandler: ChannelDuplexHandler {
         var headers = request.headers
 
         if !request.headers.contains(name: "Host") {
-            headers.add(name: "Host", value: request.host)
+            let port = request.port
+            let host = (port != 80 && port != 443) ? "\(request.host):\(port)" : request.host
```

Comment:
shouldn't we only exclude `80` if we're speaking (plain text) HTTP and only exclude `443` when we're speaking HTTPS?

Say I wanted to do `http://localhost:443` (note the missing `s` in `http`), then I should get `localhost:443` or am I missing something? CC @Lukasa ?

Reply:
From RFC7230 "If the port is equal to the default port for a scheme, the normal form is to omit the port subcomponent"
I guess 443 is not the normal port, for `http` so it should not be submitted. Do we want to support the other way. ie `https://locahost:80/`?

Reply:
I think so. I think the only instances where we can omit the port numbers are

- `http://...:80`
- `https://...:443`

Reply:
done

Reply:
swift format is wanting me to output it as 
```
if port != 80 || request.scheme != "http", port != 443 || request.scheme != "https" {
```
instead of 
```
if (port != 80 || request.scheme != "http") && (port != 443 || request.scheme != "https") {
```
Is that what you are expecting?

Reply:
I really don't like swift-format messing with those things but I guess the argument is consistency ...

Reply:
@adam-fowler I think we're discussing on the wrong line here :)

Reply:
I meant applying [De Morgan](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) to the condition so switching from

       (port != 80 || request.scheme != "http") && (port != 443 || request.scheme != "https")

to

    ! ((port == 80 && request.scheme == "http") || (port == 443 && request.scheme == "https"))

but it's personal preference